src/main/scala-2.12/com/linagora/gatling/imap/action/ImapResponseHandler.scala

```diff
@@ -25,8 +23,8 @@ class ImapResponseHandler(imapActionContext: ImapActionContext, requestName: Str
 
   override def receive: Receive = {
     case Response(responses) =>
-      val (checkSaveUpdate, error) = Check.check(responses, session, checks.toList)
-      error.fold(ok(session, start, checkSaveUpdate))(ko(session, start))
+      val (newSession, error) = Check.check(responses, session, checks.toList)
```

Comment:
could someone check this specific part, I'm not sure I understand it correctly

Reply:
seems good to me